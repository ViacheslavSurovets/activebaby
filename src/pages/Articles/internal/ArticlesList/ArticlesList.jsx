import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectArticlesForOverview } from '@redux/articles/articles.selectors';
import { ArticleListItemComponent } from '../ArticleListitemComponent';
import PropTypes from 'prop-types';


const ArticlesList = ( { articles } ) => {

  return (
    <>
      {
        articles.map (  article =>
          <Fragment key={ article.id }>
            <ArticleListItemComponent article={ article } />
          </Fragment>
        )
      }
    </>
  );
};

const mapStateToProps = createStructuredSelector ( {
  articles: selectArticlesForOverview
} );

export default connect ( mapStateToProps ) ( ArticlesList );

ArticlesList.propTypes = {
  articles: PropTypes.array
};

