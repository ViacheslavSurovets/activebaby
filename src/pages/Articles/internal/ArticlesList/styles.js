import styled from 'styled-components';

export const ArticleListContainer = styled.div`
  width: 100%;
  margin-top: 5rem;
`;

