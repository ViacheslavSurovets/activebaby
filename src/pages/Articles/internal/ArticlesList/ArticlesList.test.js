import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import ArticlesList from './ArticlesList';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('ArticlesList', ()=> {
  it('ArticlesList Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <ArticlesList  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
