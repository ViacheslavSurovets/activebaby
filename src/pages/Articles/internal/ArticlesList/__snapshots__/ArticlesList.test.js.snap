// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList ArticlesList Snapshot 1`] = `ShallowWrapper {}`;
